api/src/main/kotlin/org/jetbrains/kotlinx/dl/api/core/layer/pooling/MaxPool1D.kt

```diff
@@ -0,0 +1,109 @@
+/*
+* Copyright 2020 JetBrains s.r.o. and Kotlin Deep Learning project contributors. All Rights Reserved.
+* Use of this source code is governed by the Apache 2.0 license that can be found in the LICENSE.txt file.
+*/
+
+package org.jetbrains.kotlinx.dl.api.core.layer.pooling
+
+import org.jetbrains.kotlinx.dl.api.core.KGraph
+import org.jetbrains.kotlinx.dl.api.core.layer.Layer
+import org.jetbrains.kotlinx.dl.api.core.layer.convolutional.ConvPadding
+import org.jetbrains.kotlinx.dl.api.core.shape.convOutputLength
+import org.tensorflow.Operand
+import org.tensorflow.Shape
+import org.tensorflow.op.Ops
+import org.tensorflow.op.core.Squeeze
+
+/**
+ * Max pooling operation for 1D temporal data (e.g. audio, timeseries).
+ *
+ * Downsamples the input by taking maximum value over a temporal window of size [poolSize].
+ *
+ * @property [poolSize] Size of the temporal pooling window for each dimension of input.
+ * @property [strides] The amount of shift for pooling window in each pooling step. If
+ * `null`, it will default to [poolSize].
+ * @property [padding] Padding strategy; can be either of [ConvPadding.VALID] which means no padding, or
+ * [ConvPadding.SAME] which means padding the input equally such that the output has the same dimension
+ * as the input.
+ */
+public class MaxPool1D(
+    public val poolSize: LongArray = longArrayOf(1, 2, 1),
+    public val strides: LongArray? = null,
```

Comment:
Sorry, but we need to avoid nulls in constructors and hidden dependency and logic between input parameters, please add default value here 1, 2, 1